sklearn/model_selection/_validation.py

```diff
@@ -279,25 +382,61 @@ def _fit_and_score(estimator, X, y, scorer, train, test, verbose,
     return ret
```

Comment:
So maybe ret should be a dict? it's a variable length list now. I guess it's not that bad.